File: backend/app/api/v1/analytics.py
from fastapi import APIRouter, Depends, HTTPException, status, Query
from typing import List, Optional
from decimal import Decimal
from datetime import date, timedelta
from app.schemas.analytics import (
    SalesOverview,
    SalesByPeriod,
    CreditNotesByPeriod,
    DailySalesTrend,
    InventorySummary,
    InventoryPaymentStatus,
    LowStockAlert,
    StockMovementSummary,
    CustomerSalesAnalysis,
    CustomerPaymentAnalysis,
    TopCustomer,
    TopSellingProduct,
    SalesByCategory,
    PaymentCollectionRate,
    ExpenseSummary,
    ExpensesByPeriod,
    ExpensesByCategory,
    OutgoingPaymentsSummary,
    OutgoingExpensePayments,
    CreditNoteRefunds,
    IncomingPaymentsSummary,
    DashboardSummary
)
from app.api.deps import get_current_user, get_current_company, require_admin
from app.utils.supabase import get_supabase
from supabase import Client

router = APIRouter(dependencies=[Depends(require_admin)])

# ==========================================
# SALES ANALYTICS
# ==========================================

@router.get("/sales/overview", response_model=SalesOverview)
async def get_sales_overview(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("sales_overview")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return SalesOverview(
                company_id=company["id"],
                total_invoices=0,
                total_invoice_amount=Decimal("0"),
                paid_invoices=0,
                paid_invoice_amount=Decimal("0"),
                unpaid_invoices=0,
                unpaid_invoice_amount=Decimal("0"),
                partial_invoices=0,
                partial_invoice_amount=Decimal("0"),
                partial_amount_paid=Decimal("0"),
                partial_amount_due=Decimal("0"),
                total_outstanding=Decimal("0"),
                total_returns=0,
                total_return_amount=Decimal("0"),
                net_sales=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/sales/by-period", response_model=SalesByPeriod)
async def get_sales_by_period(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("sales_by_period")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return SalesByPeriod(
                company_id=company["id"],
                today_sales=Decimal("0"),
                today_invoice_count=0,
                week_sales=Decimal("0"),
                week_invoice_count=0,
                month_sales=Decimal("0"),
                month_invoice_count=0,
                year_sales=Decimal("0"),
                year_invoice_count=0
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/sales/credit-notes-by-period", response_model=CreditNotesByPeriod)
async def get_credit_notes_by_period(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("credit_notes_by_period")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return CreditNotesByPeriod(
                company_id=company["id"],
                today_returns=Decimal("0"),
                week_returns=Decimal("0"),
                month_returns=Decimal("0"),
                year_returns=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/sales/daily-trend", response_model=List[DailySalesTrend])
async def get_daily_sales_trend(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase),
    days: int = Query(30, ge=1, le=365)
):
    try:
        response = supabase.table("daily_sales_trend")\
            .select("*")\
            .eq("company_id", company["id"])\
            .order("sale_date", desc=True)\
            .limit(days)\
            .execute()

        return response.data
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


# ==========================================
# INVENTORY ANALYTICS
# ==========================================

@router.get("/inventory/summary", response_model=InventorySummary)
async def get_inventory_summary(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("inventory_summary")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return InventorySummary(
                company_id=company["id"],
                total_variants=0,
                total_units=0,
                total_inventory_value_cost=Decimal("0"),
                total_inventory_value_retail=Decimal("0"),
                potential_profit=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/inventory/payment-status", response_model=InventoryPaymentStatus)
async def get_inventory_payment_status(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("inventory_payment_status")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return InventoryPaymentStatus(
                company_id=company["id"],
                paid_stock_transactions=0,
                paid_stock_value=Decimal("0"),
                unpaid_stock_transactions=0,
                unpaid_stock_value=Decimal("0"),
                partial_stock_transactions=0,
                partial_stock_value=Decimal("0"),
                partial_amount_paid=Decimal("0"),
                total_outstanding=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/inventory/low-stock", response_model=List[LowStockAlert])
async def get_low_stock_alerts(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("low_stock_alerts")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        return response.data
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/inventory/stock-movement", response_model=List[StockMovementSummary])
async def get_stock_movement(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("stock_movement_summary")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        return response.data
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


# ==========================================
# CUSTOMER ANALYTICS
# ==========================================

@router.get("/customers/sales-analysis", response_model=List[CustomerSalesAnalysis])
async def get_customer_sales_analysis(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase),
    limit: int = Query(100, le=500)
):
    try:
        response = supabase.table("customer_sales_analysis")\
            .select("*")\
            .eq("company_id", company["id"])\
            .limit(limit)\
            .execute()

        return response.data
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/customers/payment-analysis", response_model=List[CustomerPaymentAnalysis])
async def get_customer_payment_analysis(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase),
    limit: int = Query(100, le=500)
):
    try:
        response = supabase.table("customer_payment_analysis")\
            .select("*")\
            .eq("company_id", company["id"])\
            .limit(limit)\
            .execute()

        return response.data
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/customers/top-customers", response_model=List[TopCustomer])
async def get_top_customers(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase),
    limit: int = Query(10, le=50)
):
    try:
        response = supabase.table("top_customers_by_sales")\
            .select("*")\
            .eq("company_id", company["id"])\
            .limit(limit)\
            .execute()

        return response.data
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


# ==========================================
# PRODUCT ANALYTICS
# ==========================================

@router.get("/products/top-selling", response_model=List[TopSellingProduct])
async def get_top_selling_products(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase),
    limit: int = Query(10, le=50)
):
    try:
        response = supabase.table("top_selling_products")\
            .select("*")\
            .eq("company_id", company["id"])\
            .limit(limit)\
            .execute()

        return response.data
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/products/sales-by-category", response_model=List[SalesByCategory])
async def get_sales_by_category(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("sales_by_category")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        return response.data
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


# ==========================================
# PAYMENT ANALYTICS
# ==========================================

@router.get("/payments/collection-rate", response_model=PaymentCollectionRate)
async def get_payment_collection_rate(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("payment_collection_rate")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return PaymentCollectionRate(
                company_id=company["id"],
                total_invoices=0,
                total_billed=Decimal("0"),
                total_collected=Decimal("0"),
                collection_rate_percentage=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


# ==========================================
# EXPENSE ANALYTICS
# ==========================================

@router.get("/expenses/summary", response_model=ExpenseSummary)
async def get_expense_summary(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("expense_summary")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return ExpenseSummary(
                company_id=company["id"],
                total_standard_expenses=0,
                total_standard_amount=Decimal("0"),
                total_sales_expenses=0,
                total_sales_amount=Decimal("0"),
                total_expenses=0,
                total_amount=Decimal("0"),
                total_paid=Decimal("0"),
                total_outstanding=Decimal("0"),
                unpaid_count=0,
                unpaid_amount=Decimal("0"),
                partial_count=0,
                partial_outstanding=Decimal("0"),
                paid_count=0,
                paid_amount=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/expenses/by-period", response_model=ExpensesByPeriod)
async def get_expenses_by_period(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("expenses_by_period")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return ExpensesByPeriod(
                company_id=company["id"],
                today_expenses=Decimal("0"),
                today_count=0,
                week_expenses=Decimal("0"),
                week_count=0,
                month_expenses=Decimal("0"),
                month_count=0,
                year_expenses=Decimal("0"),
                year_count=0
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/expenses/by-category", response_model=List[ExpensesByCategory])
async def get_expenses_by_category(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("expenses_by_category")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        return response.data
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


# ==========================================
# OUTGOING PAYMENTS
# ==========================================

@router.get("/payments/outgoing-inventory", response_model=OutgoingPaymentsSummary)
async def get_outgoing_inventory_payments(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("outgoing_payments_summary")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return OutgoingPaymentsSummary(
                company_id=company["id"],
                inventory_outstanding=Decimal("0"),
                inventory_paid_this_month=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/payments/outgoing-expenses", response_model=OutgoingExpensePayments)
async def get_outgoing_expense_payments(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("outgoing_expense_payments")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return OutgoingExpensePayments(
                company_id=company["id"],
                expense_outstanding=Decimal("0"),
                expense_paid_this_month=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


@router.get("/payments/credit-note-refunds", response_model=CreditNoteRefunds)
async def get_credit_note_refunds(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("credit_note_refunds")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return CreditNoteRefunds(
                company_id=company["id"],
                refunds_outstanding=Decimal("0"),
                refunds_paid_this_month=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


# ==========================================
# INCOMING PAYMENTS
# ==========================================

@router.get("/payments/incoming", response_model=IncomingPaymentsSummary)
async def get_incoming_payments(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        response = supabase.table("incoming_payments_summary")\
            .select("*")\
            .eq("company_id", company["id"])\
            .execute()

        if not response.data or len(response.data) == 0:
            return IncomingPaymentsSummary(
                company_id=company["id"],
                unpaid_invoice_count=0,
                unpaid_invoice_amount=Decimal("0"),
                partial_invoice_count=0,
                partial_invoice_amount=Decimal("0"),
                total_pending_incoming=Decimal("0")
            )

        return response.data[0]
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))


# ==========================================
# DASHBOARD SUMMARY
# ==========================================

@router.get("/dashboard", response_model=DashboardSummary)
async def get_dashboard_summary(
    current_user = Depends(get_current_user),
    company = Depends(get_current_company),
    supabase: Client = Depends(get_supabase)
):
    try:
        sales_by_period = await get_sales_by_period(current_user, company, supabase)
        credit_notes_by_period = await get_credit_notes_by_period(current_user, company, supabase)
        expenses_by_period = await get_expenses_by_period(current_user, company, supabase)
        outgoing_inventory = await get_outgoing_inventory_payments(current_user, company, supabase)
        outgoing_expenses = await get_outgoing_expense_payments(current_user, company, supabase)
        outgoing_credit_notes = await get_credit_note_refunds(current_user, company, supabase)
        incoming_payments = await get_incoming_payments(current_user, company, supabase)
        sales_overview = await get_sales_overview(current_user, company, supabase)
        inventory_summary = await get_inventory_summary(current_user, company, supabase)
        low_stock = await get_low_stock_alerts(current_user, company, supabase)
        top_customers = await get_top_customers(current_user, company, supabase, limit=5)
        top_products = await get_top_selling_products(current_user, company, supabase, limit=5)
        collection_rate = await get_payment_collection_rate(current_user, company, supabase)
        expense_summary = await get_expense_summary(current_user, company, supabase)

        return DashboardSummary(
            sales_by_period=sales_by_period,
            credit_notes_by_period=credit_notes_by_period,
            expenses_by_period=expenses_by_period,
            outgoing_inventory=outgoing_inventory,
            outgoing_expenses=outgoing_expenses,
            outgoing_credit_notes=outgoing_credit_notes,
            incoming_payments=incoming_payments,
            sales_overview=sales_overview,
            inventory_summary=inventory_summary,
            low_stock_count=len(low_stock),
            top_customers=top_customers,
            top_products=top_products,
            payment_collection_rate=collection_rate,
            expense_summary=expense_summary
        )
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))

File: frontend/src/lib/analytics.ts
import { apiClient } from './axios';

// ==========================================
// TYPES
// ==========================================

export interface SalesOverview {
  company_id: string;
  total_invoices: number;
  total_invoice_amount: number;
  paid_invoices: number;
  paid_invoice_amount: number;
  unpaid_invoices: number;
  unpaid_invoice_amount: number;
  partial_invoices: number;
  partial_invoice_amount: number;
  partial_amount_paid: number;
  partial_amount_due: number;
  total_outstanding: number;
  total_returns: number;
  total_return_amount: number;
  net_sales: number;
}

export interface SalesByPeriod {
  company_id: string;
  today_sales: number;
  today_invoice_count: number;
  week_sales: number;
  week_invoice_count: number;
  month_sales: number;
  month_invoice_count: number;
  year_sales: number;
  year_invoice_count: number;
}

export interface CreditNotesByPeriod {
  company_id: string;
  today_returns: number;
  week_returns: number;
  month_returns: number;
  year_returns: number;
}

export interface DailySalesTrend {
  company_id: string;
  sale_date: string;
  invoice_count: number;
  daily_sales: number;
  daily_returns: number;
  net_daily_sales: number;
}

export interface InventorySummary {
  company_id: string;
  total_variants: number;
  total_units: number;
  total_inventory_value_cost: number;
  total_inventory_value_retail: number;
  potential_profit: number;
}

export interface LowStockAlert {
  company_id: string;
  variant_id: string;
  product_id: string;
  product_name: string;
  variant_name: string;
  sku: string | null;
  location_name: string;
  current_stock: number;
  reorder_threshold: number;
  units_below_threshold: number;
  unit_cost: number;
  estimated_reorder_cost: number;
  stock_status: string;
}

export interface TopCustomer {
  company_id: string;
  customer_id: string;
  customer_name: string;
  customer_type: string;
  total_sales: number;
  invoice_count: number;
  average_order_value: number | null;
}

export interface TopSellingProduct {
  company_id: string;
  variant_id: string;
  product_id: string;
  product_name: string;
  variant_name: string;
  sku: string | null;
  category_name: string | null;
  total_quantity_sold: number;
  total_revenue: number;
  number_of_sales: number;
  average_selling_price: number;
  total_cost: number;
  gross_profit: number;
}

export interface PaymentCollectionRate {
  company_id: string;
  total_invoices: number;
  total_billed: number;
  total_collected: number;
  collection_rate_percentage: number;
}

export interface ExpenseSummary {
  company_id: string;
  total_standard_expenses: number;
  total_standard_amount: number;
  total_sales_expenses: number;
  total_sales_amount: number;
  total_expenses: number;
  total_amount: number;
  total_paid: number;
  total_outstanding: number;
  unpaid_count: number;
  unpaid_amount: number;
  partial_count: number;
  partial_outstanding: number;
  paid_count: number;
  paid_amount: number;
}

export interface ExpensesByPeriod {
  company_id: string;
  today_expenses: number;
  today_count: number;
  week_expenses: number;
  week_count: number;
  month_expenses: number;
  month_count: number;
  year_expenses: number;
  year_count: number;
}

export interface ExpensesByCategory {
  company_id: string;
  category_id: string;
  category_name: string;
  expense_type: string;
  expense_count: number;
  total_amount: number;
  total_paid: number;
  total_outstanding: number;
}

export interface OutgoingPaymentsSummary {
  company_id: string;
  inventory_outstanding: number;
  inventory_paid_this_month: number;
}

export interface OutgoingExpensePayments {
  company_id: string;
  expense_outstanding: number;
  expense_paid_this_month: number;
}

export interface CreditNoteRefunds {
  company_id: string;
  refunds_outstanding: number;
  refunds_paid_this_month: number;
}

export interface IncomingPaymentsSummary {
  company_id: string;
  unpaid_invoice_count: number;
  unpaid_invoice_amount: number;
  partial_invoice_count: number;
  partial_invoice_amount: number;
  total_pending_incoming: number;
}

export interface DashboardSummary {
  // Row 1 - Sales
  sales_by_period: SalesByPeriod | null;
  credit_notes_by_period: CreditNotesByPeriod | null;
  // Row 2 - Expenses
  expenses_by_period: ExpensesByPeriod | null;
  // Row 3 - Outgoing payments
  outgoing_inventory: OutgoingPaymentsSummary | null;
  outgoing_expenses: OutgoingExpensePayments | null;
  outgoing_credit_notes: CreditNoteRefunds | null;
  // Row 4 - Incoming payments
  incoming_payments: IncomingPaymentsSummary | null;
  // Supporting data
  sales_overview: SalesOverview | null;
  inventory_summary: InventorySummary | null;
  low_stock_count: number;
  top_customers: TopCustomer[];
  top_products: TopSellingProduct[];
  payment_collection_rate: PaymentCollectionRate | null;
  expense_summary: ExpenseSummary | null;
}

// ==========================================
// API FUNCTIONS
// ==========================================

export const analyticsAPI = {
  getDashboard: async (): Promise<DashboardSummary> => {
    const response = await apiClient.get<DashboardSummary>('/analytics/dashboard/');
    return response.data;
  },
  getSalesOverview: async (): Promise<SalesOverview> => {
    const response = await apiClient.get<SalesOverview>('/analytics/sales/overview/');
    return response.data;
  },
  getSalesByPeriod: async (): Promise<SalesByPeriod> => {
    const response = await apiClient.get<SalesByPeriod>('/analytics/sales/by-period/');
    return response.data;
  },
  getCreditNotesByPeriod: async (): Promise<CreditNotesByPeriod> => {
    const response = await apiClient.get<CreditNotesByPeriod>('/analytics/sales/credit-notes-by-period/');
    return response.data;
  },
  getDailySalesTrend: async (days: number = 30): Promise<DailySalesTrend[]> => {
    const response = await apiClient.get<DailySalesTrend[]>('/analytics/sales/daily-trend/', { params: { days } });
    return response.data;
  },
  getInventorySummary: async (): Promise<InventorySummary> => {
    const response = await apiClient.get<InventorySummary>('/analytics/inventory/summary/');
    return response.data;
  },
  getLowStockAlerts: async (): Promise<LowStockAlert[]> => {
    const response = await apiClient.get<LowStockAlert[]>('/analytics/inventory/low-stock/');
    return response.data;
  },
  getTopCustomers: async (limit: number = 10): Promise<TopCustomer[]> => {
    const response = await apiClient.get<TopCustomer[]>('/analytics/customers/top-customers/', { params: { limit } });
    return response.data;
  },
  getTopProducts: async (limit: number = 10): Promise<TopSellingProduct[]> => {
    const response = await apiClient.get<TopSellingProduct[]>('/analytics/products/top-selling/', { params: { limit } });
    return response.data;
  },
  getPaymentCollectionRate: async (): Promise<PaymentCollectionRate> => {
    const response = await apiClient.get<PaymentCollectionRate>('/analytics/payments/collection-rate/');
    return response.data;
  },
  getExpenseSummary: async (): Promise<ExpenseSummary> => {
    const response = await apiClient.get<ExpenseSummary>('/analytics/expenses/summary/');
    return response.data;
  },
  getExpensesByPeriod: async (): Promise<ExpensesByPeriod> => {
    const response = await apiClient.get<ExpensesByPeriod>('/analytics/expenses/by-period/');
    return response.data;
  },
  getExpensesByCategory: async (): Promise<ExpensesByCategory[]> => {
    const response = await apiClient.get<ExpensesByCategory[]>('/analytics/expenses/by-category/');
    return response.data;
  },
  getOutgoingInventoryPayments: async (): Promise<OutgoingPaymentsSummary> => {
    const response = await apiClient.get<OutgoingPaymentsSummary>('/analytics/payments/outgoing-inventory/');
    return response.data;
  },
  getOutgoingExpensePayments: async (): Promise<OutgoingExpensePayments> => {
    const response = await apiClient.get<OutgoingExpensePayments>('/analytics/payments/outgoing-expenses/');
    return response.data;
  },
  getCreditNoteRefunds: async (): Promise<CreditNoteRefunds> => {
    const response = await apiClient.get<CreditNoteRefunds>('/analytics/payments/credit-note-refunds/');
    return response.data;
  },
  getIncomingPayments: async (): Promise<IncomingPaymentsSummary> => {
    const response = await apiClient.get<IncomingPaymentsSummary>('/analytics/payments/incoming/');
    return response.data;
  },
};